apis/r/R/Init.R

```diff
@@ -0,0 +1,25 @@
+## .onLoad is called whether code from the package is used and the packages is 'loaded'. An
+## example is calling `tiledbsoma::show_package_versions()`. So this is most elementary check,
+## .onAttach is also called when the package is 'attached' via 'library(tiledbsoma)'
+## During package build and byte-code compilation and load check, both are called.
+.onLoad <- function(libname, pkgname) {
+    rpkg_lib_version <- tiledb::tiledb_version(compact=TRUE)
+    soma_lib_version <- libtiledbsoma_version(compact=TRUE)
+    if (rpkg_lib_version != soma_lib_version) {
+        msg <- sprintf("TileDB Core version %s used by TileDB-R package, but TileDB-SOMA uses %s",
+                       sQuote(rpkg_lib_version), sQuote(soma_lib_version))
+        stop(msg, call. = FALSE)
+    }
+}
+
+## A _possible_ .onAttach() function -- these are not allowed to use cat() etc but _must_ communicate
+## via packageStartupMessage() as this function can be 'muzzled' as desired. See Writing R Extensions.
+## .onAttach <- function(libname, pkgname) {
+##     if (interactive()) {
+##         packageStartupMessage("TileDB-SOMA R package ", packageVersion(pkgname),
+##                               " with TileDB Embedded ", format(tiledb::tiledb_version(TRUE)),
+##                               " on ", utils::osVersion,
+##                               ".\nSee https://github.com/single-cell-data for more information ",
+##                               "about the SOMA project.")
+##     }
+## }
```

Comment:
Please remove, or uncomment and utilize

Reply:
I knew this would come but I kept it _on purpose_:  it is near impossible to test or reason about .onAttach and .onLoad without having a working example.  (Illustration: this one is hard to test via a unit test, so you may want to check out the package and flip the `!=` on line eight to a `==` just to see.  It is also useful to see if the test could be in .onAttach (tl;dr: not really).  So give it another minute of thought.  And while you are there, we could also contemplate actually using the .onAttach message just like tiledb-r does:

```r
> library(tiledbsoma)
TileDB-SOMA R package 1.9.99.3 with TileDB Embedded 2.23.0 on Ubuntu 23.10.
See https://github.com/single-cell-data for more information about the SOMA project.
> 
```

But because this is more-or-less outside of the scope of this PR I did not feel like forcing it in.  We could though.

Thoughts?  Easy to adjust either way.  My weak vote is on uncommenting .onAttach.  If you feel that needs another PR it can of course be done.

Reply:
@eddelbuettel I've accepted the PR, with advice -- feel free to merge at will

Reply:
@johnkerl Good. I didn't mean to or want to 'impose' or 'force' the added note but I think of it as an improvement so I will add it with a second commit.  If it is found to be a nuisance it is equally easy to remove.